presenters/workspace_manager_presenter.py

```diff
@@ -68,7 +74,15 @@ def _save_selected_workspace(self):
             return
         selected_workspace = selected_workspaces[0]
         path = self._workspace_manger_view.get_workspace_to_save_filepath()
-        self._work_spaceprovider.save_nexus(selected_workspace, path)
+        if not path:
+            self._workspace_manger_view.error_invalid_save_path()
+            return
+        if not path.endswith('.nxs'):
+            path += '.nxs'
+        try:
+            self._work_spaceprovider.save_nexus(selected_workspace, path)
+        except:
```

Comment:
It's considered bad style to catch all exceptions. I think `RuntimeError` would be most sensible here too.